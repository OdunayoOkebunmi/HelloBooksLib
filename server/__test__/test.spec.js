// eslint-disable-next-line import/no-extraneous-dependencies
import supertest from 'supertest';
import app from '../index';

const server = () => supertest(app);
const url = '/api/v1';

describe('Welcome to helo books', () => {
  it('Should return a message on app start', done => {
    server()
      .get(`${url}/`)
      .end((err, res) => {
        expect(res.statusCode).toEqual(200);
        expect(res.body.message).toEqual('This is Hello books');
        done();
        expect(res.body).toMatchSnapshot();
      });
  });

  it('Should return an error message for an invalid URL', done => {
    server()
      .get(`${url}/unknown`)
      .end((err, res) => {
        expect(res.statusCode).toEqual(404);
        expect(res.body.error).toEqual(
          'Sorry, the requested endpoint does not exist on our server'
        );
        done();
        expect(res.body).toMatchSnapshot();
      });
  });

  it('Should redirect to /api/v1', done => {
    server()
      .get(`/`)
      .end((err, res) => {
        expect(res.statusCode).toEqual(302);
        done();
        expect(res.body).toMatchSnapshot();
      });
  });
});
