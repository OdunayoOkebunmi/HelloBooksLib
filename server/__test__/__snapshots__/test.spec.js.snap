// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome to helo books Should return a message on app start 1`] = `
Object {
  "message": "This is Hello books",
}
`;

exports[`Welcome to helo books Should return an error message for an invalid URL 1`] = `
Object {
  "error": "Sorry, the requested endpoint does not exist on our server",
}
`;
